core/src/main/java/org/apache/gravitino/job/BuiltInJobTemplateEventListener.java

```
@@ -0,0 +1,395 @@
+            .filter(entity -> isValidBuiltInName(entity.name()))
+            .collect(Collectors.toMap(JobTemplateEntity::name, entity -> entity));
+
+    builtInTemplates.forEach(
+        (name, newTemplate) -> {
+          JobTemplateEntity existing = existingBuiltIns.remove(name);
+          if (existing == null) {
+            registerNewBuiltInJobTemplate(metalake, newTemplate);
+          } else {
+            int existingVersion = version(existing.templateContent().customFields());
+            int newVersion = version(newTemplate.customFields());
+            if (newVersion > existingVersion) {
+              updateBuiltInJobTemplate(metalake, existing, newTemplate);
+            } else {
+              LOG.info("Built-in job template {} under metalake {} is up to date", name, metalake);
+            }
+          }
+        });
+
+    existingBuiltIns.values().forEach(entity -> deleteObsoleteBuiltInJobTemplate(metalake, entity));
+  }
+
+  private void registerNewBuiltInJobTemplate(String metalake, JobTemplate jobTemplate) {
+    JobTemplateEntity jobTemplateEntity =
+        JobTemplateEntity.builder()
+            .withId(idGenerator.nextId())
+            .withName(jobTemplate.name())
+            .withNamespace(NamespaceUtil.ofJobTemplate(metalake))
+            .withComment(jobTemplate.comment())
+            .withTemplateContent(JobTemplateEntity.TemplateContent.fromJobTemplate(jobTemplate))
+            .withAuditInfo(
+                AuditInfo.builder()
+                    .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                    .withCreateTime(Instant.now())
+                    .build())
+            .build();
+
+    try {
+      jobManager.registerJobTemplate(metalake, jobTemplateEntity);
+      LOG.info(
+          "Registered built-in job template {} under metalake {}", jobTemplate.name(), metalake);
+    } catch (JobTemplateAlreadyExistsException e) {
+      LOG.warn(
```

Do you think we should propagate the exception here? Since the job template is expected not to exist.